### PR TITLE
Image Editor: When editing, pass the edited image src.

### LINKED ISSUE #23702: Image src may not match attachment ID
**Describe the bug**
When editing an image, the "Save" request relies on the attachment ID in the image block. However in some edge cases that ID may not match the edited image's attachment. For example after all posts have been exported from one website and imported in another.

Then, if the wrong attachment ID is not for an image, the Save request fails, but if the wrong ID is for another image's attachment, the other image is edited and replaced in the image block.

**To reproduce**
1. Upload two images and add them to a post.
2. From the Code editor replace the ID on the first image (in the block's args and in the `wp-image-nnnn` class) with the ID for the second. Save and preview the post, notice the first image doesn't have `srcset`, etc, as the attachment ID is wrong.
3. Edit the first image and save. At this point the edits are applied to the second image, and the first image is replaced with the second.

To fix it will need to use the new `wp_image_file_matches_image_meta()` on the API side to confirm the src and attachment ID are for the same image.

Opened https://core.trac.wordpress.org/ticket/50565 for trunk.

### PR DESCRIPTION
**Pull request by azaozz**
Fixes https://github.com/WordPress/gutenberg/issues/23702.

## Description
Adds the image `src` to the Save request when editing an image. On the API side the image meta (retrieved by using the attachment ID) is verified against the src.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Comment by azaozz** [comment]
Corresponding trac ticket: https://core.trac.wordpress.org/ticket/50565.

**Review by TimothyBJacobs — APPROVED** [review]
I think we can safely merge this now. The REST API doesn't error on unrecognized parameters, so if someone is running on Trunk + GB Trunk the parameter will be passed and just won't be operated on.

We'll need to wait to merge the patch in Core until the same time as the updated GB JS lands.

**Comment by azaozz** [comment]
Needs to be added to trunk (ideally before 5.5-beta2) to unblock https://core.trac.wordpress.org/ticket/50565. Only the tiny js change matters :)

**Pull request merged**
